Test certificate creation tool

asymmetric opened the pull request:
Closes giantswarm/aws-operator#18.

Commit:
Support certificate authorization for Kubernetes

Support TLS authorization in the Kubernetes client based on
authority certificate, client certificate and client key.

Fixes #10

Commit:
make: Add check target and include testify in vendor

First of all, this change introduces a "make check" target
which runs unit tests.

The other change is using glide-vc with --use-glide-lock options
which is the only way to prevent the removal of the testify library
(because it's not being detected by "glide list" which is used in
glide-vc by default).

Commit:
glide: Add testify library

Testify provides a clean way of doing assertions in unit tests.

Commit:
vendor: Update vendor

Commit:
Make aws-operator open sourced under Apache License 2.0

Provide necessary files to clarify the license of the project,
maintainers, ways to contact with the community and the goals.

iaguis committed:
aws: create instance only if it doesn't exist or is terminated (#19)

After stoping or terminating instances, they stay in stopped or
terminated state for a while.

Only if they don't exist or are terminated, we should create the
instance.

iaguis committed:
vendor: fix typo on testify's repo name (#21)

* vendor: fix typo on testify's repo name

And move it to testImport.

* vendor: update deps

iaguis committed:
Refactor allExistingInstancesMatch (#20)

* Refactor allExistingInstancesMatch

* Add test for allExistingInstancesMatch

asymmetric committed:
ignore certs directory

asymmetric committed:
add tool to create test certs

asymmetric committed:
force dir creation

asymmetric committed:
comments

asymmetric committed:
decrease key size

asymmetric committed:
fix broken loop

asymmetric committed:
hardcode algo

asymmetric committed:
move cert generation in for loop

Commit:
cloudconfig: Use k8scloudconfig in create operator

We moved code for cloudconfig generation to k8scloudconfig library
which provides the common template for cloudconfigs and allows
to inject operator-specific things.

Fixes #23

Commit:
make: Add update-bindata target

This target allows to generate bindata.go module from the files
which we store in templates/ directory.

Commit:
templates: Add templates for decrypt-tls-assets service

This service will be used for fetching keys from AWS to the
instance.

Commit:
bindata: Update bindata

Commit:
glide: Update/add dependencies

The following dependencies were updated:
- giantswarm/awstpr
- fsnotify/fsnotify
- pelletier/go-toml
- spf13/cobra

The following dependencies were added:
- giantswarm/clustertpr
- giantswarm/k8scloudconfig
giantswamr/giant

Commit:
vendor: Update vendor

Commit:
create: Use separated awstpr and clustertpr

Fixes #24

Commit:
examples: Adjust cluster yaml example to new tprs

Commit:
glide: Update versions of awstpr and clustertpr

Commit:
vendor: Update vendor

nhlfr commented:
LGTM, but I think you should submit a PR for giantswarm repo, not kinvolk's fork. ;)

iaguis commented:
And you can squash the commits

JosephSalisbury committed:
Changes circle to build with architect (#26)

asymmetric committed:
use genrsa and improve comments

JosephSalisbury committed:
Changes circle to fetch architect unauthed if token not present

We usually fetch architect's latest binary authenticated, to
avoid any issues with rate limiting.
However, we don't pass secrets to forked builds, for security
reasons, so forked builds can't fetch the architect binary at all.

This changeset makes the fetch fall back to unauthenticated if the
token is not present - this means that forked builds work, but may
be subjected to GitHub rate limiting.

Commit:
create: Pass the certificates to KMS

Fixes #17

Commit:
glide: Update dependencies

Commit:
vendor: Update vendor

iaguis committed:
create: pass instance profile to new instances (#34)

If we don't do it, the instance won't have permissions to decrypt the
certificates.

iaguis committed:
service: put policy/role code in a better place (#35)

It has nothing to do with encoding TLS assets.

This prepares the code for uploading cloudconfig to S3.

iaguis committed:
create: fix typos in the assume role policy (#37)

iaguis committed:
create: make instance creation more robust

* Wait until the instance profile exists before continuing
* Retry instance creation because Roles take some time to propagate to
instance profiles

iaguis committed:
vendor: bump k8scloudconfig

And update vendoring.

Commit:
glide: Update awstpr

Commit:
vendor: Update vendor

asymmetric committed:
use CN from parameter

iaguis committed:
create: don't continue if there's an error (#42)

If there's an error which is an AWS error but is not
"Invalid IAM Instance Profile", we should return an error and not
continue.

iaguis committed:
service: rework roles/policies/instance profiles (#39)

* service: rework roles/policies/instance profiles

We now name roles, policies and instance profiles in a per-cluster
basis.

Also, we treat already-existing roles or instance profiles as errors.
Reusing them doesn't really work. In any case, they should've been
removed when we deleted the cluster (for now we need to do it manually
since the delete operator is not implemented yet).

* create: use same name for instance profile and role

So we can delete both when deleting the role from the aws console:

"You cannot use the console to delete an instance profile, except when
it has the exact same name as the role and you delete it as part of the
process of deleting a role as described in the preceding procedure."

http://docs.aws.amazon.com/IAM/latest/UserGuide/id_roles_manage_delete.html#roles-managingrole-deleting-console

asymmetric commented:
Closed in favor of #44.
